### PR TITLE
week1_vinayak

### PR DESCRIPTION
**Pull request by Vinayak-Cruzer**
Snap of HTML Page
html_snap_iris

### CODE REVIEW

**Review by Vinayak-Cruzer — COMMENTED** [review]
changing to return  from puts inside the function armstrong_number(n)

File: week1/armstrong.rb [hunk 1]
```diff
@@ -7,7 +7,41 @@
...
-def armstrong_number(n)
-    raise NotImplementedError # Delete this line and start coding here
-end
+def pow(a,b)
+    result = 1
+    b.times do
+        result=result*a
+    end
+    return result
+  end
```
Comment (raosush) [inline week1/armstrong.rb:16]: Ruby's USP is it's elegance and optimisations for programmer's happiness 🙂
For `pow`, you can just do:
```ruby
x**t
```
Which is equivalent to x^t

File: week1/armstrong.rb [hunk 2]
```diff
@@ -7,7 +7,41 @@
...
+  def armstrong_number(n)
+    total_digits=0
+    digit=0
+    num=n
+    while(num>0)
+      #digit=num%10
+      num/=10
+      total_digits=total_digits+1
+    end
```
Comment (raosush) [inline week1/armstrong.rb:25]: Digits are easy to find:
```ruby
array_of_digits = num.digits
```

File: week1/armstrong.rb [hunk 3]
```diff
@@ -7,7 +7,41 @@
...
+    i=1
```
Comment (raosush) [inline week1/armstrong.rb:26]: Maybe just use `n>0` condition

File: week1/armstrong.rb [hunk 4]
```diff
@@ -7,7 +7,41 @@
...
+    end
+  if (number==sum)
+      return "true"
+  else
+      return "false"
+  end
+  
+  end
+  
+  n=gets.chomp().to_i
```
Comment (raosush) [inline week1/armstrong.rb:43]: Our testing script directly invokes the function. This is unnecessary and also the reason for the check failing

**Review by raosush — CHANGES_REQUESTED** [review]
Assignment completed, but you could remove the `gets ` call to get the checks passing

**Pull request closed without merging**